mock/channels/index.html

```diff
@@ -1,73 +1,138 @@
+          case false:
+            const testChannel = await joinChannel(context.joinAppChannel);
+            await broadcast(context, testChannel);
+            addCloseWindowListener(testChannel);
+            break;
+          case true:
+            await broadcast(context);
+            await joinChannel(context.joinAppChannel);
+            addCloseWindowListener();
+            break;
         }
+      }
 
-        //Channels app joins channel 1
-        const joinChannel = async () => {
-          const channels = await window.fdc3.getSystemChannels();
-          if (channels.length > 0) {
-            try {
-              await window.fdc3.joinChannel(channels[0].id);
-            } catch (ex) {
-              throw new Error("Error while calling joinChannel in app B: " + (ex.message ?? ex))
-            }
-          } else {
-            throw new Error("No system channels available for app B");
+      const addCloseWindowListener = async (testChannel) => {
+        if (testChannel === undefined) {
+          await window.fdc3.addContextListener(
+            "closeWindow",
+            () => {
+              closeFinsembleWindow();
+            });
+        } else {
+          await testChannel.addContextListener(
+            "closeWindow",
+            () => {
+              closeFinsembleWindow();
+            });
+        }
+      }
+
+      const joinChannel = async (useAppChannel) => {
+        if (useAppChannel) {
+          return await joinAppChannel();
+        } else {
+          await joinSystemChannel();
+        }
+      }
+
+      const broadcast = async (context, testChannel) => {
+        if (testChannel === undefined) {
+          await systemBroadcast(context.contextBroadcasts);
+        } else {
+          await appChannelBroadcast(context, testChannel);
+        }
+      }
+
+      //Channels app joins channel 1
+      const joinSystemChannel = async () => {
+        const channels = await window.fdc3.getSystemChannels();
+        if (channels.length > 0) {
+          try {
+            await window.fdc3.joinChannel(channels[0].id);
+            document.getElementById("context").innerHTML += `Joined user channel ${channels[0].id}/ `;
+          } catch (ex) {
+            throw new Error("Error while calling joinChannel in app B: " + (ex.message ?? ex))
           }
+        } else {
+          throw new Error("No system channels available for app B");
         }
+      }
 
-        //Broadcast from channels app
-        const broadcast = async (contextBroadcasts) => {
-          for (const systemBroadcast in contextBroadcasts) {
-            if (contextBroadcasts[systemBroadcast]) {
-              await window.fdc3.broadcast({
-                "type": `fdc3.${systemBroadcast}`
-              });
+      //Channels app joins test-channel
+      const joinAppChannel = async () => {
+        try {
+          var appChannel = await fdc3.getOrCreateChannel("test-channel");
```

Comment:
Can use const instead of var

Reply:
Not sure why I did that. I must have momentarily lapsed into c# without realising